packages/core-js/internals/fix-regexp-well-known-symbol-logic.js

```diff
@@ -5,24 +5,33 @@ var fails = require('../internals/fails');
 var requireObjectCoercible = require('../internals/require-object-coercible');
 var wellKnownSymbol = require('../internals/well-known-symbol');
 
-module.exports = function (KEY, length, exec) {
+module.exports = function (KEY, length, exec, sham) {
   var SYMBOL = wellKnownSymbol(KEY);
   var methods = exec(requireObjectCoercible, SYMBOL, ''[KEY]);
   var stringMethod = methods[0];
   var regexMethod = methods[1];
   if (fails(function () {
+    // String methods call symbol-named RegEp methods
     var O = {};
     O[SYMBOL] = function () { return 7; };
     return ''[KEY](O) != 7;
+  }) || fails(function () {
+    // Symbol-named RegExp methods call .exec
```

Comment:
This test throws an uncaught stack overflow error in the promises tests and I'm not sure why.